tangelo/problem_decomposition/dmet/fragment.py

```diff
@@ -63,6 +64,11 @@ def __post_init__(self):
         self.fermionic_hamiltonian = self._get_fermionic_hamiltonian()
         self.frozen_mos = None
 
+    @property
+    def __class__(self):
+        # Tricking isinstance function.
+        return SecondQuantizedMolecule
```

Comment:
I do not like this, because we instantiated a fragment object and now regardless of where it's used, querying for type will return `SecondQuantizedMolecule`. This could lead to sneaky bugs in the future with DMET.

I could not find any reference on when to overwrite the `__class__` method: have you found anything in PEP or in search engines ?

If this object can be considered a `SecondQuantizedMolecule` with a few extra functionalities and fields, then maybe this class should inherit from `SecondQuantizedMolecule` and we could redefine post_init / init. In that situation, we wouldn't even have to fool `isInstance`, as it would return `True`.

It's tempting to just merge, but I just want to have more clarity on the matter. If this is a pain in the butt, well, we may move forward with this solution.

Reply:
I change the check with the options we discussed via zoom.